addons/account/models/account_move.py

```diff
@@ -2208,7 +2208,7 @@ def copy(self, default=None):
         return copied_am
 
     def _sanitize_vals(self, vals):
-        if 'invoice_line_ids' in vals and 'line_ids' in vals:
+        if 'invoice_line_ids' in vals and 'line_ids' in vals and bool(vals['invoice_line_ids']) and bool(vals['line_ids']):
```

Comment:
To make the lines shorter and easier to read
```suggestion
        if vals.get('invoice_line_ids') and vals.get('line_ids'):
```

Reply:
Thank you for your guidance, I have learned a lot from you